lib/tasks/translations/transifex_updater.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+class TransifexUpdater
+  def update
+    return unless updatable_locales.present? && tx_client_present?
+
+    fetch_translations
+    update_translations
+  end
+
+  private
+
+  def updatable_locales
+    locales = ENV.fetch("AVAILABLE_LOCALES", "").split(",").collect(&:strip)
+    locales.delete("en")
+    locales
+  end
+
+  def tx_client_present?
+    log "Transifex Client:"
+    system("tx --version")
+  end
+
+  # Fetches translations for available locales that have changed since the local git version.
+  # Any updated translations are saved in a separate file (eg `fr.yml.new`) and the original
+  # file (eg `fr.yml`) is left untouched.
+  def fetch_translations
+    system("tx pull --use-git-timestamps --disable-overwrite -l #{updatable_locales.join(',')}")
+  end
+
+  # Processes each new translation file (if any) and applies updates to original
+  def update_translations
+    return unless new_translation_files.any?
+
+    new_translation_files.each do |new_file|
+      current_file = new_file.chomp('.new')
+      updated_translations = apply_updates(current_file, new_file)
+
+      File.open(current_file, "w") do |file|
+        file.write updated_translations.to_yaml(line_width: -1)
```

Comment:
Ruby's `to_yaml` doesn't quote strings by default, only when necessary. That leads to a much larger diff. I got over 2000 line changes for one locale. I couldn't find any option to change this.

Transifex seems to preserve the quoting of the original YAML file. So the only solution to this would be to to use the same formatting as `to_yaml` produces in the source `en.yml` file. Omitting quotes can lead to more mistakes editing the file by hand though. I'm not sure if it's worth it.